src/main/java/com/ghostipedia/cosmiccore/api/recipe/lookup/MapHeatIngredient.java

```diff
@@ -0,0 +1,25 @@
+package com.ghostipedia.cosmiccore.api.recipe.lookup;
+
+import com.gregtechceu.gtceu.api.recipe.lookup.AbstractMapIngredient;
+
+public class MapHeatIngredient extends AbstractMapIngredient {
```

Comment:
you don't need/want this. only items and fluids (and other "stacks" of stuff) need Map ingredients.